src/sagemaker/jumpstart/types.py

```diff
@@ -1732,32 +1732,15 @@ def from_json(self, json_obj: Dict[str, Any]) -> None:
         )
         self.inference_config_rankings: Optional[Dict[str, JumpStartConfigRanking]] = (
             {
-                alias: JumpStartConfigRanking(ranking)
+                alias: JumpStartConfigRanking(ranking, is_hub_content=self._is_hub_content)
                 for alias, ranking in json_obj["inference_config_rankings"].items()
             }
             if json_obj.get("inference_config_rankings")
             else None
         )
-        inference_configs_dict: Optional[Dict[str, JumpStartMetadataConfig]] = (
-            {
-                alias: JumpStartMetadataConfig(
-                    alias,
-                    config,
-                    json_obj,
-                    (
-                        {
-                            component_name: self.inference_config_components.get(component_name)
-                            for component_name in config.get("component_names")
-                        }
-                        if config and config.get("component_names")
-                        else None
-                    ),
-                )
-                for alias, config in json_obj["inference_configs"].items()
-            }
-            if json_obj.get("inference_configs")
-            else None
-        )
+        inference_configs_dict: Optional[Dict[str, JumpStartMetadataConfig]] = json_obj[
```

Comment:
I'm not sure if this works, now that the `inference_configs_dict` becomes a json object, how would we construct the `JumpStartMetadataConfig` class without parsing the json?

Reply:
+1, also curious if we should do `.get` here also to mitigate future errors